rtnetlink/src/route/add.rs

```diff
@@ -0,0 +1,153 @@
+use futures::stream::StreamExt;
+use std::net::IpAddr;
+
+use netlink_packet_route::{
+    nlas::route::Nla, NetlinkMessage, NetlinkPayload, RouteKind, RouteMessage, RouteProtocol,
+    RouteScope, RouteTable, RtnlMessage, AF_INET, AF_INET6, NLM_F_ACK, NLM_F_CREATE, NLM_F_EXCL,
+    NLM_F_REQUEST,
+};
+
+use crate::{Error, ErrorKind, Handle};
+
+fn addr_octets(addr: &IpAddr) -> Vec<u8> {
+    match addr {
+        IpAddr::V4(addr) => addr.octets().to_vec(),
+        IpAddr::V6(addr) => addr.octets().to_vec(),
+    }
+}
+
+/// A request to create a new route. This is equivalent to the `ip route add` commands.
+pub struct RouteAddRequest {
+    handle: Handle,
+    message: RouteMessage,
+}
+
+impl RouteAddRequest {
+    pub(crate) fn new(handle: Handle) -> Self {
+        let mut message = RouteMessage::default();
+
+        message.header.table = RouteTable::Main;
+        message.header.protocol = RouteProtocol::Static;
+        message.header.scope = RouteScope::Universe;
+        message.header.kind = RouteKind::Unicast;
+
+        RouteAddRequest { handle, message }
+    }
+
+    fn set_addr_family(&mut self, addr: &IpAddr) {
```

Comment:
I would prefer `set_address_family`

Reply:
Also why are we passing an `IpAddr`? Can't we just pass a constant, either `AF_INET` or `AF_INET6`?